core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -781,6 +796,318 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
+            action: WhiskAction,
+            //env: Option[Parameters],
+            payload: Option[JsObject],
+            blocking: Boolean,
+            topmost: Boolean,
+            components: Vector[FullyQualifiedEntityName],
+            cause: Option[ActivationId],
+            atomicActionsCount: Int)(implicit transid: TransactionId): Future[(ActivationId, Option[WhiskActivation], Int)] = {
+        // create new activation id that corresponds to the sequence
+        val seqActivationId = activationIdFactory.make()
+        info(this, s"Invoking sequence $action topmost $topmost activationid '$seqActivationId'")
+        val start = Instant.now(Clock.systemUTC())
+        var seqActivation: Option[WhiskActivation] = None
+        // the cause for the component activations is the current sequence
+        val futureWskActivations = invokeSequenceComponents(user, action, seqActivationId, payload, components, cause = Some(seqActivationId), atomicActionsCount)
+        val futureSeqResult = Future.sequence(futureWskActivations)
+        val response: Future[(ActivationId, Option[WhiskActivation], Int)] =
+            if (topmost) { // need to deal with blocking and closing connection
+                if (blocking) {
+                    val timeout = maxWaitForBlockingActivation + blockingInvokeGrace
+                    val futureSeqResultTimeout = futureSeqResult withTimeout(timeout, new BlockingInvokeTimeout(seqActivationId))
+                    // if the future fails with a timeout, the failure is dealt with at the caller level
+                    futureSeqResultTimeout map { wskActivationTuples =>
+                        val wskActivationEithers = wskActivationTuples.map(_._1)
+                        // the execution of the sequence was successful, return the result
+                        val end = Instant.now(Clock.systemUTC())
+                        seqActivation = Some(makeSequenceActivation(user, action, seqActivationId, wskActivationEithers, topmost, cause, start, end))
+                        val atomicActionCnt = wskActivationTuples.last._2
+                        (seqActivationId, seqActivation, atomicActionCnt)
+                    }
+                } else {
+                    // non-blocking sequence execution, return activation id
+                    Future.successful((seqActivationId, None, 0))
+                }
+            } else {
+                // not topmost, no need to worry about terminating incoming request
+                futureSeqResult map { wskActivationTuples =>
+                     val wskActivationEithers = wskActivationTuples.map(_._1)
+                    // all activations are successful, the result of the sequence is the result of the last activation
+                    val end = Instant.now(Clock.systemUTC())
+                    seqActivation = Some(makeSequenceActivation(user, action, seqActivationId, wskActivationEithers, topmost, cause, start, end))
```

Comment:
makeSequenceActivation could throw an exception and this will cause the future to complete with an error and not create an activation.